pkg/provider/aws/execution_unit.go

```diff
@@ -154,23 +154,26 @@ func (a *AWS) handleHelmChartAwsValues(chart *kubernetes.HelmChart, unit *core.E
 			}
 			valueParams[val.Key] = resources.TargetGroupCreateParams{
 				AppName: a.Config.AppName,
-				Refs:    core.AnnotationKeySetOf(unit.AnnotationKey),
-				Name:    unit.ID,
+				Refs:    core.BaseConstructSetOf(unit),
+				Name:    unit.Name,
 			}
 		}
 	}
 	return
 }
 
-func (a *AWS) getLambdaConfiguration(result *core.ConstructGraph, dag *core.ResourceGraph, refs core.AnnotationKeySet) (resources.LambdaFunctionConfigureParams, error) {
-	ref, oneRef := refs.GetSingle()
-	if !oneRef {
+func (a *AWS) getLambdaConfiguration(result *core.ConstructGraph, dag *core.ResourceGraph, refs core.BaseConstructSet) (resources.LambdaFunctionConfigureParams, error) {
+	if len(refs) > 1 {
 		return resources.LambdaFunctionConfigureParams{}, fmt.Errorf("lambda must only have one construct reference")
 	}
+	var ref core.BaseConstruct
+	for r := range refs {
+		ref = r
+	}
```

Comment:
won't this result in a panic if there are 0 refs? (same comment applies in multiple places)

Reply:
I dont think it panics, but we may have an error, the original code was if there [wasnt a single ref](https://github.com/klothoplatform/klotho/blob/3c7f578fe18977a913132ef57bb795736052162a/pkg/provider/aws/execution_unit.go#L167), so i need the first condition to check if there is one ref or 2.

Reply:
oh we would panic, i see because ref would be nil and we would try to call nil on it

Reply:
just pushed an update for this

Reply:
It won't panic, but it will leave `ref` as the zero-value.

Reply:
realistically this whole logic will be moved into the compiler ux and these will be converted into constraints, but not a reason to not have it work now